### ISSUE TITLE
mgrid with complex step generates inaccurate results

### ISSUE THREAD

**Issue by Armin-Ariamajd** [google/jax#13782]
### Description

Using `jnp.mgrid` with negative `start` and complex `step` generates floats with large errors, e.g.:

```python
>>> import jax.numpy as jnp
>>> jnp.mgrid[-4:-1:4j]
Array([-4., -2.9999998, -2. , -1.], dtype=float32)
>>> jnp.mgrid[-3:0:4j]
Array([-3., -1.9999999 , -0.99999994,  0.], dtype=float32)
```

### What jax/jaxlib version are you using?

jax v0.4.1, jaxlib v0.4.1

### Which accelerator(s) are you using?

CPU

### Additional system info

Python 3.10.8, macOS 12.6.1

**Comment by jakevdp** [comment]
Thanks for the report. This comes from the implementation of `linspace`:
```python
import jax.numpy as jnp
print(jnp.linspace(-4, -1, 4))
# [-4.        -2.9999998 -2.        -1.       ]
```
Note that this output is accurate to about one part in $10^7$, which is typical of 32-bit floating-point arithmetic (`float32` has 23 bits of mantissa, and $1/2^{23} \sim 10^{-7}$). With that in mind, I think we can label this as working as intended.

If you need more accuracy than is provided by float32, you can enable 64-bit computation in JAX following the instructions here: https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision

**Comment by jakevdp** [comment]
Closing because I don't think there are any action items here.
